contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+    uint256 private _nextWithdrawIdx;
+
+    uint256 private _bnbxHoldingBalance; // amount of bnbx held by this strategy
+    uint256 private _bnbDepositBalance; // amount of bnb deposited by this strategy
+    uint256 private _bnbxToUnstake; // amount of bnbx to withdraw from stader in next batchWithdraw
+    uint256 private _bnbToDistribute; // amount of bnb to distribute to users who unstaked
+
+    event StakeManagerChanged(address stakeManager);
+
+    /// @dev initialize function - Constructor for Upgradable contract, can be only called once during deployment
+    /// @param destination Address of the stakeManager contract
+    /// @param feeRecipient Address of the fee recipient, which receives yield
+    /// @param bnbxToken Address of BNBx token
+    /// @param masterVault Address of the masterVault contract
+    /// @param stakeManager Address of stakeManager contract
+    function initialize(
+        address destination,
+        address feeRecipient,
+        address bnbxToken,
+        address masterVault,
+        address stakeManager
+    ) public initializer {
+        __BaseStrategy_init(destination, feeRecipient);
+
+        _bnbxToken = IERC20Upgradeable(bnbxToken);
+        _stakeManager = IStakeManager(stakeManager);
+        _vault = IMasterVault(masterVault);
+
+        _bnbxToken.approve(stakeManager, type(uint256).max);
+    }
+
+    /**
+     * Modifiers
+     */
+    modifier onlyVault() {
+        require(msg.sender == address(_vault), "!vault");
+        _;
+    }
+
+    /// @dev deposits the given amount of BNB into Stader stakeManager through bnbxRouter
+    function deposit() external payable onlyVault returns (uint256) {
+        uint256 amount = msg.value;
+        require(amount <= address(this).balance, "insufficient balance");
+
+        return _deposit(amount);
+    }
+
+    /// @dev deposits all the available BNB into Stader stakeManager through bnbxRouter
+    function depositAll() external payable onlyVault returns (uint256) {
+        return _deposit(address(this).balance);
+    }
+
+    /// @dev internal function to deposit the given amount of BNB into Stader stakeManager through bnbxRouter
+    /// @param amount amount of BNB to deposit
+    function _deposit(uint256 amount)
+        internal
+        whenDepositNotPaused
+        returns (uint256)
+    {
+        require(canDeposit(amount), "invalid amount");
+
+        _bnbDepositBalance += amount;
+        _bnbxHoldingBalance += _stakeManager.convertBnbToBnbX(amount);
+        _stakeManager.deposit{value: amount}();
+        return amount;
+    }
+
+    /// @dev withdraws the given amount of BNB from Stader's stakeManager
+    /// @param amount amount of BNB to withdraw
+    function withdraw(address recipient, uint256 amount)
+        external
+        onlyVault
+        returns (uint256)
+    {
+        return _withdraw(recipient, amount);
+    }
+
+    /// @dev withdraws everything from Stader's stakeManager
```

Comment:
Can you elaborate on `everything`

Reply:
pstake and Helio [discussion](https://github.com/helio-money/helio-smart-contracts/pull/9#discussion_r1013989015) over this.
everything = bnb in strategy `(users' unstaked funds + extraBNB if any)` + bnb still deposited in stakeManager

And as per master vault's code, it is `bnbDepositBalance` i.e.  bnb still deposited in stakeManager

Reply:
Per Ankr's integration code, users "unstaked" funds will never be panic-withdrawable.